### PR TITLE
New Example: Vis and Filter Integration

### PR DESCRIPTION
**Pull request by dbchristopher**
This PR creates a new extension example to illustrate how Vis and Filter components can be used together within the extension framework.

## Screenshot:

Screen Shot 2022-04-19 at 3 16 01 PM

### CODE REVIEW

File: react/typescript/vis-and-filter-components/README.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+   application: vis-filter-integration {
+    label: "Visualization and Filter Components (TypeScript)"
+    url: "http://localhost:8080/bundle.js"
+      entitlements: {
+         core_api_methods: ["all_connections", "all_looks", "create_query", "dashboard", "lookml_model_explore", "query_for_slug", "query", "run_inline_query", "run_look", "run_query", "search_folders", "model_fieldname_suggestions"]
+      }
+   }
+   ```
+
+6. Create a `model` LookML file in your project. The name doesn't matter but the convention is to name it the same as the project— in this case, helloworld-js.
```
Comment (bryans99) [inline react/typescript/vis-and-filter-components/README.md:59]: should be
> in this case, vis-filter-integration

File: react/typescript/vis-and-filter-components/README.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+9. Reload the page and click the `Browse` dropdown menu. You will see the extension in the list.
+
+- The extension will load the JavaScript from the `url` provided in the `application` definition. By default, this is http://localhost:8080/bundle.js. If you change the port your server runs on in the package.json, you will need to also update it in the manifest.lkml.
+- Refreshing the extension page will bring in any new code changes from the extension template, although some changes will hot reload.
+
+## Extension Entitlements
+
+Entitlements are defined in the project manifest file for the extension.
+
+Resources required by the extension (Looker API methods for example) must be defined in entitlements. This extension uses the `me` api method, as such it is defined in the entitlements.
```
Comment (bryans99) [inline react/typescript/vis-and-filter-components/README.md:81]: This refers to the me api call. Paragraph needs a little rework

File: react/typescript/vis-and-filter-components/README.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    file: "bundle.js"
+      entitlements: {
+         core_api_methods: ["all_connections", "all_looks", "create_query", "dashboard", "lookml_model_explore", "query_for_slug", "query", "run_inline_query", "run_look", "run_query", "search_folders", "model_fieldname_suggestions"]
+      }
+   }
+   ```
+
+## Notes
+
+- Webpack's module splitting is not currently supported.
```
Comment (bryans99) [inline react/typescript/vis-and-filter-components/README.md:117]: No longer true. Remove (I need to remove that everywhere)

File: react/typescript/vis-and-filter-components/manifest.lkml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+project_name: "helloworld-ts"
```
Comment (bryans99) [inline react/typescript/vis-and-filter-components/manifest.lkml:1]: Sample manifest needs fixing

File: react/typescript/vis-and-filter-components/package.json [hunk 5]
```diff
@@ -0,0 +1,149 @@
+{
+  "name": "vis-and-filter-components",
+  "version": "1.1.0",
+  "description": "Looker Extension SDK Hello World TypeScript Demo",
```
Comment (bryans99) [inline react/typescript/vis-and-filter-components/package.json:4]: Description needs fixing

File: react/typescript/vis-and-filter-components/src/App.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+ */
+import React from 'react'
+import { ExtensionProvider } from '@looker/extension-sdk-react'
+import { hot } from 'react-hot-loader/root'
+import { ComponentsProvider } from '@looker/components'
+
+import { VisualizationComponents } from './VisualizationComponents'
+
+export const App = hot(() => (
```
Comment (bryans99) [inline react/typescript/vis-and-filter-components/src/App.tsx:33]: hot not needed - can be removed

File: react/typescript/vis-and-filter-components/src/VisualizationComponents.tsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+  return (
+    <DataProvider sdk={core40SDK}>
+      <FieldText
+        label="Query (Numeric ID or Slug)"
+        name="query"
+        value={queryIdentifier}
+        onChange={handleQueryIdChange}
+        my="xsmall"
+        placeholder="abcxyz..."
```
Comment (bryans99) [inline react/typescript/vis-and-filter-components/src/VisualizationComponents.tsx:56]: placeholder?

**Review by bryans99 — CHANGES_REQUESTED** [review]
Found some minor stuff.

Will take it for a spin tomorrow

**Comment by dbchristopher** [comment]
closing in favor of #83. I'll apply your feedback there @bryans99

File: react/typescript/vis-and-filter-components/src/VisualizationComponents.tsx (hunk 7 above)
Comment (dbchristopher) [reply react/typescript/vis-and-filter-components/src/VisualizationComponents.tsx:56]: lol the placehcolder was just meant to represent the random assortment of letters and numbers that make up a query slug. I can see how that's not very clear though.

**Pull request closed without merging**